packages/theia-extension/cdtcloud/src/browser/cdtcloud-widget.tsx

```diff
@@ -25,53 +40,65 @@ export class CdtcloudWidget extends ReactWidget {
     this.title.closable = true;
     this.title.iconClass = 'fa fa-window-maximize';
     this.update();
+    this.getDeviceList();
   }
 
+  Selector = () => {
+    let [board, setBoard] = React.useState({ label: "No board Selected", value: "" });
+    return (<div>
+      <Select
+        value={board}
+        options={this.options}
+        onChange={e => {
+          if (!e) return
+          const newBoard = { label: e.label, value: e.value }
+          setBoard(newBoard);
+        }}
+      />
+      <button className='theia-button secondary' title='Display Message' onClick={_a => this.deployOnBoard(board)}>Deploy on Board</button>
+    </div>)
+  }
 
 
-
-  render(): React.ReactElement {
-
+  render(): React.ReactElement<any, string | React.JSXElementConstructor<any>> {
     const header = `This widget enables you to deploy your code on a remote (Arduino-)board.`;
-
-    //hier availableDevices einfügen
-    const options = [
-      { value: 'uniqueId1', label: 'Arduino Uno' },
-      { value: 'uniqueId2', label: 'Arduino Due' },
-      { value: 'uniqueId3', label: 'Arduino Trio' }
-    ]
-
+    
     return <div id='widget-container'>
       <AlertMessage type='INFO' header={header} />
 
       <h2> Select a Board to deploy your code on from this list</h2>
-
-      <Select options={options} />
-      <button className='theia-button secondary' title='Display Message' onClick={_a => this.deployOnBoard()}>Deploy on Board</button>
+      <this.Selector />
     </div>
   }
 
-  protected async deployOnBoard(): Promise<void> {
-
-    const data = { deviceTypeId: '37dcb3ab-071d-43e1-935a-1e70403e4720', artifactUri: 'https://sanctum-dev.com/tests.bin' };
+  protected handleChange(option: { label: string; value: string }): void {
+    this.selected = option
+  }
 
+  protected async getDeviceList(): Promise<void> {
     try {
-      await fetch('http://127.0.0.1:3001/deployments', {
-        method: 'POST',
-        headers: {
-          'Content-Type': 'application/json',
-        },
-        body: JSON.stringify(data),
-      });
+      this.deviceList = await this.deviceTypeService.getDeviceList()
+      this.options = this.deviceList.map(({ id, name }) => ({ label: name, value: id }))
+      this.update()
+    } catch (err) {
+      console.log(err)
     }
-    catch { }
-
-    this.messageService.info('Deployment Request send.');
-  }
-
-
 
 
+  }
 
+  protected async deployOnBoard(board: any): Promise<void> {
+    const selectedBoard = this.deviceList.find(obj => {
+      return obj.id === board.value
+    })
+    const workspaceService = this.workspaceService;
+    let sketchPath = "" + workspaceService.workspace?.resource.path
+    sketchPath = sketchPath.substring(sketchPath.indexOf('/') + 1);
+    console.log(sketchPath)
+    if (sketchPath) {
+      this.compilationService.compile(selectedBoard.fqbn, board.value, sketchPath)
+    }
 
-}
+    this.messageService.info('Deployment Request send.');
+  }
```

Comment:
There are going to be problems with the paths. Arduino-Cli expects a total path. The removed root `/` causes issues here. Because of the specific types of `Uri` and `Path` in theia, we can't resolve this with the normal methods. We could import the `FileUri` namespace here to get the correct path.


```suggestion
    const sketchUri = this.workspaceService.workspace?.resource
        if (sketchUri === undefined) {
          throw new Error('No Sketch found')
        }

        const sketchPath = FileUri.fsPath(sketchUri)
        await this.compilationService.compile(selectedBoard.fqbn, board.value, sketchPath)

        await this.messageService.info('Deployment Request send.');
      }
```